### PR TITLE
add aggressiveness to uturn

### PR DESCRIPTION
**Pull request by iman512003**
This is to address #279 .
Screenshot from 2020-12-21 17-20-28

### CODE REVIEW

**Review by iman512003 — COMMENTED** [review]
I still need to fix sensor test

File: smarts/core/mission_planner.py [hunk 1]
```diff
@@ -277,11 +282,60 @@ def cut_in_waypoints(self, sim, pose: Pose, vehicle):
...
+            vehicle=vehicle, radius=200
         )
 
+        if not neighborhood_vehicles:
+            return []
+
+        aggressiveness = self._agent_behavior.aggressiveness
+        if sim.elapsed_sim_time < 0.5:
+            return []
+        if sim.elapsed_sim_time == 0.5:
```
Comment (liamchzh) [inline smarts/core/mission_planner.py:295]: We shouldn't rely on `elapsed_sim_time == 0.5`

File: smarts/core/mission_planner.py [hunk 2]
```diff
@@ -45,13 +45,18 @@ def __init__(
...
         self._waypoints = waypoints
+        self._agent_behavior = agent_behavior
         self._mission = None
         self._route = None
         self._road_network = road_network
         self._did_plan = False
         self._task_is_triggered = False
         self._uturn_initial_heading = 0
-        self._agent_behavior = agent_behavior
+        self._uturn_initial_distant = 0
```
[the remark this answers is not in the document]
Comment (liamchzh) [reply smarts/core/mission_planner.py:55]: No need to address now.

File: examples/ego_open_agent.py [hunk 3]
```diff
@@ -21,15 +21,16 @@
 
 
 def main(scenarios, headless, num_episodes, seed):
-    open_agent_spec = open_agent.entrypoint(debug=False)
+    open_agent_spec = open_agent.entrypoint(debug=False, aggressiveness=0.1)
```
[the remark this answers is not in the document]
Comment (iman512003) [reply examples/ego_open_agent.py]: Thanks,

File: smarts/core/mission_planner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iman512003) [reply smarts/core/mission_planner.py:55]: Thanks, I put a todo in the comments and created #346 .

File: smarts/core/mission_planner.py (hunk 1 above)
Comment (iman512003) [reply smarts/core/mission_planner.py:295]: Thanks,

**Review by liamchzh — APPROVED** [review]

**Pull request merged**